### ISSUE TITLE
Indicator area overflow over the bar or slider area in v9

### ISSUE THREAD

**Issue by rmshub** [kisvegabor/lvgl#5187]
### Introduce the problem

Code to reproduce the issue
### Bar
``` 
lv_obj_t *prog = lv_bar_create(design_panel);
lv_obj_set_size(prog, LV_PCT(94), LV_PCT(15));
lv_bar_set_value(prog, 0, LV_ANIM_ON);
lv_obj_set_style_radius(prog, 0, LV_PART_INDICATOR);
lv_obj_set_style_radius(prog, 5, 0);
lv_bar_set_range(prog, 0, 100);
```
### Slider
```
lv_obj_t *slider = lv_slider_create(brightness);
lv_obj_set_size(slider, 400, 50);
lv_slider_set_range(slider, 0, 100);
lv_slider_set_value(slider, lcd_brightness, LV_ANIM_ON);
lv_obj_remove_style(slider, NULL, LV_PART_KNOB); /* Be sure the knob is not displayed */
lv_obj_add_flag(slider, LV_OBJ_FLAG_ADV_HITTEST);
lv_obj_set_style_radius(slider, 10, LV_PART_MAIN | LV_STATE_DEFAULT);
lv_obj_set_style_radius(slider, 0, LV_PART_INDICATOR);
```
Branch: master
![image](https://github.com/lvgl/lvgl/assets/38435057/d35ab0e7-dc95-4942-9478-281797b8cf1d)
![image](https://github.com/lvgl/lvgl/assets/38435057/216de0fe-092f-4fbc-aa66-8fa5c06b61c2)
![image](https://github.com/lvgl/lvgl/assets/38435057/1352ee2e-9f5f-4c59-88c9-4595507f281e)

**Comment by hyansuper** [comment]
@rmshub  Do you mean indicator overflow the rounded corner? but you set it so

> lv_obj_set_style_radius(prog, 0, LV_PART_INDICATOR);
lv_obj_set_style_radius(prog, 5, 0);

you can remove the fist line

**Comment by rmshub** [comment]
Yes, indicator overflow at rounded corner. It is happen in v9. Same code doesn't generate overflow at corner in v8.3.11.  I want indicator as straight line while moving.

**Comment by kisvegabor** [comment]
I'm working on it and will send a PR still today.

**Comment by kisvegabor** [comment]
Should be fixed by #5229. Let's continue in the PR.
